components/button/button-add.js

```diff
@@ -110,18 +92,91 @@ class ButtonAdd extends PropertyRequiredMixin(FocusMixin(LocalizeCoreElement(Lit
-				<div class="content">
-					<d2l-icon icon="tier1:plus-default"></d2l-icon>
-					${this.textVisible
-		? html`<span>${text}</span>`
-		: html`<d2l-tooltip class="vdiff-target" offset="18" for="${this._buttonId}" for-type="label">${text}</d2l-tooltip>`}
-				</div>
 			</button>
 		`;
 	}
+
+	_renderWithTextHidden(text) {
+		return html`
+			<d2l-button-add-icon-text ?visible-on-ancestor="${this.iconOnlyVisibleOnHoverFocus}"></d2l-button-add-icon-text>
+			<d2l-tooltip class="vdiff-target" offset="18" for="${this._buttonId}" for-type="label">${text}</d2l-tooltip>
+		`;
+	}
+
 }
 customElements.define('d2l-button-add', ButtonAdd);
 
+/**
+ * @ignore
+ */
+class ButtonAddIconText extends VisibleOnAncestorMixin(LitElement) {
+	static get properties() {
+		return {
+			text: { type: String }
+		};
+	}
+
+	static get styles() {
+		return [visibleOnAncestorStyles, css`
+			:host {
+				--d2l-button-add-icon-text-color: var(--d2l-color-galena);
+				align-items: center;
+				display: flex;
+			}
+			:host([text]) {
+				--d2l-button-add-icon-text-color: var(--d2l-color-celestine);
+				color: var(--d2l-button-add-icon-text-color);
+				height: 1.5rem;
+				padding: 0 0.3rem;
+			}
+
+			svg {
+				fill: var(--d2l-button-add-icon-text-color);
+			}
+
+			:host([text]) svg {
+				padding-inline-end: 0.2rem;
+			}
+			:host(:not([text])) svg {
+				margin: -3px; /** hover/click target */
+				padding: 3px; /** hover/click target */
+			}
+
+			span {
+				font-size: 0.7rem;
+				font-weight: 700;
+				letter-spacing: 0.2px;
+				line-height: 1rem;
+			}`
+		];
+	}
+
+	render() {
+		return html`
+			<svg width="18" height="18" viewBox="0 0 18 18" xmlns="http://www.w3.org/2000/svg">
+				<g>
+					<path fill-rule="evenodd" clip-rule="evenodd" d="M9 2C10.3845 2 11.7378 2.41054 12.889 3.17971C14.0401 3.94888 14.9373 5.04213 15.4672 6.32122C15.997 7.6003 16.1356 9.00777 15.8655 10.3656C15.5954 11.7235 14.9287 12.9708 13.9497 13.9497C12.9708 14.9287 11.7235 15.5954 10.3656 15.8655C9.00777 16.1356 7.6003 15.997 6.32122 15.4672C5.04213 14.9373 3.94888 14.0401 3.17971 12.889C2.41054 11.7378 2 10.3845 2 9C2.00212 7.14413 2.7403 5.36489 4.05259 4.05259C5.36489 2.7403 7.14413 2.00212 9 2V2ZM9 0C6.61305 0 4.32387 0.948212 2.63604 2.63604C0.948212 4.32387 0 6.61305 0 9C0 11.3869 0.948212 13.6761 2.63604 15.364C4.32387 17.0518 6.61305 18 9 18C11.3869 18 13.6761 17.0518 15.364 15.364C17.0518 13.6761 18 11.3869 18 9C18 6.61305 17.0518 4.32387 15.364 2.63604C13.6761 0.948212 11.3869 0 9 0V0Z" />
+					<path fill-rule="evenodd" clip-rule="evenodd" d="M10 5C10 4.44772 9.55228 4 9 4C8.44772 4 8 4.44772 8 5V8H5C4.44772 8 4 8.44772 4 9C4 9.55228 4.44772 10 5 10H8V13C8 13.5523 8.44772 14 9 14C9.55228 14 10 13.5523 10 13V10H13C13.5523 10 14 9.55228 14 9C14 8.44772 13.5523 8 13 8H10V5Z" />
+				</g>
+			</svg>
+			${this.text ? html`<span class="d2l-label-text">${this.text}</span>` : nothing}
+		`;
+	}
+}
+customElements.define('d2l-button-add-icon-text', ButtonAddIconText);
```

Comment:
Do we intend to use this elsewhere? If not, just wondering if we should be exposing it publicly like this vs. just baking it into `d2l-button-add`.

Reply:
We don't intend to use this elsewhere. I separate this piece out in order to get `visible-on-ancestor-mixin` to work. Is there a way to use `visible-on-ancestor-mixin` within the same component? There are some animation tweaks I need to make to visible-on-ancestor in a follow up PR so I'm not opposed to pulling the functionality into `button-add` and modifying as needed to get it to work for a single component case if needed/possible.

Reply:
Currently, it looks like it starts its search (upwards) beginning with the host of the element extending the mixin ([here](https://github.com/BrightspaceUI/core/blob/main/mixins/visible-on-ancestor/visible-on-ancestor-mixin.js#L113)). Maybe it doesn't search within it's own DOM context because in theory, their could be a nested VOA element and target. That case might be pretty edgy. Maybe we could make VOA a little more flexible.